pdc/apps/auth/views.py

```diff
@@ -903,3 +910,231 @@ def destroy(self, request, *args, **kwargs):
         On success, HTTP status code is 204 and the response has no content.
         """
         return super(GroupResourcePermissionViewSet, self).destroy(request, *args, **kwargs)
+
+
+class APIResourcePermissionViewSet(StrictQueryParamMixin,
+                                   ChangeSetCreateModelMixin,
+                                   ChangeSetDestroyModelMixin,
+                                   mixins.RetrieveModelMixin,
+                                   mixins.ListModelMixin,
+                                   MultiLookupFieldMixin,
+                                   viewsets.GenericViewSet):
+    """
+    This end-point provides API resource permissions.
+    """
+    queryset = models.ResourcePermission.objects.all()
+    serializer_class = serializers.ResourcePermissionSerializer
+    permission_classes = (APIPermission,)
+    lookup_fields = (
+        ('resource__name', r'[^/]+'),
+        ('permission__name', r'[^/]+'),
+    )
+    filter_class = filters.ResourcePermissionFilter
+
+    def _get_users_and_groups(self, resource_permission):
```

Comment:
This method is nearly same as `get_users_and_groups`. It's not good just copy/paste to make duplicate code. E.g. if you fix read permission part later, you need to make change in 2 places rather than one. Implement it in one place make  UI  automatically consistent with API.